### PR TITLE
feat: social share

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Add og tags for social sharing, added buttons in the social drawer 

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787371/todos/4752672406)

# How should this PR be QA Tested?
- [ ] Clicking on facebook button of a published journey should take you to a new window on facebook that has a share card with the social share title description and image
- [ ] Clicking on twitter button of a published journey should take you to a new window on twitter that has a share card with the social share title description and image

# Checklist
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
Needs [this](https://github.com/JesusFilm/core/pull/485) PR to be merged in first

**Review by tataihono — CHANGES_REQUESTED** [review]
this shouldn't use javascript to open a link in a new window. Instead you should use a Material Link component with an anchor tag.

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 1]
```diff
@@ -15,6 +25,55 @@ export function SocialShareAppearance(): ReactElement {
       <ImageEdit />
       <TitleEdit />
       <DescriptionEdit />
+
+      <Typography variant="subtitle2" sx={{ pb: 4 }}>
+        Share Preview
+      </Typography>
+
+      <Stack direction="row" spacing={3}>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:33]: @tataihono should these buttons be disabled if journey is not published?

Comment (tataihono) [reply apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:33]: yes.

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 2]
```diff
@@ -15,6 +25,55 @@ export function SocialShareAppearance(): ReactElement {
...
+          startIcon={
+            <FacebookIcon
+              sx={{ height: '16px', width: '16px', color: '#1877F2' }}
+            />
+          }
+          data-testid="facebook-share-button"
+          disabled={journey == null}
+        >
+          <Link
+            component="a"
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:44]: Try convert this into a button compoent

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.spec.tsx [hunk 3]
```diff
@@ -13,5 +17,49 @@ describe('SocialShareAppearance', () => {
     expect(getByTestId('social-image-edit')).toBeInTheDocument()
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.spec.tsx:17]: Strip the test id's where possible

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 4]
```diff
@@ -15,6 +47,62 @@ export function SocialShareAppearance(): ReactElement {
...
+
+      <Typography variant="subtitle2" sx={{ pb: 4 }}>
+        Share Preview
+      </Typography>
+
+      <Stack direction="row" spacing={3}>
+        <ToolTip
+          open={openFacebookTooltip}
+          onOpen={handleOpenFacebookTooltip}
+          onClose={handleCloseFacebookTooltip}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:59]: not required

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 5]
```diff
@@ -1,11 +1,43 @@
...
+  }
+
+  const handleOpenTwitterTooltip = (): void => {
+    if (journey?.publishedAt != null) return
+    setOpenTwitterTooltip(true)
+  }
+
+  const handleCloseTwitterTooltip = (): void => {
+    setOpenTwitterTooltip(false)
+  }
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:39]: not required

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 6]
```diff
@@ -1,11 +1,43 @@
...
+import FacebookIcon from '@mui/icons-material/FacebookOutlined'
+import TwitterIcon from '@mui/icons-material/Twitter'
+import { useJourney } from '../../../../libs/context'
 import { ImageEdit } from './ImageEdit/ImageEdit'
 import { TitleEdit } from './TitleEdit/TitleEdit'
 import { DescriptionEdit } from './DescriptionEdit/DescriptionEdit'
 
 export function SocialShareAppearance(): ReactElement {
+  const journey = useJourney()
+  const shareUrl = journey?.slug ?? `untitled-journey-${journey?.id as string}`
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:16]: don't guess this.

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 7]
```diff
@@ -1,11 +1,43 @@
...
+  const encodedUrl = encodeURIComponent(shareUrl)
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:17]: why does this need to be encoded? slugs should already be ready to go into a url

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx [hunk 8]
```diff
@@ -15,6 +47,62 @@ export function SocialShareAppearance(): ReactElement {
...
+        >
+          <span>
+            <Button
+              startIcon={
+                <FacebookIcon
+                  sx={{ height: '16px', width: '16px', color: '#1877F2' }}
+                />
+              }
+              disabled={journey == null || journey?.publishedAt == null}
+              href={`https://www.facebook.com/sharer/sharer.php?u=https://your.nextstep.is/${encodedUrl}`}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/SocialShareAppearance.tsx:70]:
```suggestion
              href={`https://www.facebook.com/sharer/sharer.php?u=https://your.nextstep.is/${journey.slug}`}
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**